ONSdigital/census-rm-toolbox
# Reminder batch scheduling

# Motivation and Context
We need a way of calculating the print batches we can run each day of the reminder wave to maximize our use of the printer capacity without going over the limits, this is a script counts the matching cases for each print batch incrementally until it hit the limit and produces the action rule classifiers needed for the action rule.

# What has changed
* Add reminder batching script
* Add tests and run them in travis
* Add write context and function DB helper module and config for the RW DB
* Refactor loose python scripts into modules
* Fix flake issues with unused f-strings 

# How to test?
Build and deploy this pod, load a sample containing the treatment codes it needs (`HH_LP1E`, `HH_LP1W`, `HH_LP2E`, `HH_LP2W`), trying running the reminder batching script (follow README instructions. 
Also trying running all the other tools to check for regression after the refactor (the script commands had to be changed to use module paths).

# Links
https://trello.com/c/V3JRDv5E/977-reminders-batch-reminder-print-files-8
